pkg/controller/core/localqueue_controller.go

```diff
@@ -207,9 +211,6 @@ func (r *LocalQueueReconciler) Update(e event.UpdateEvent) bool {
 	}
 
 	r.queues.DeleteLocalQueue(oldLq)
-	if features.Enabled(features.LocalQueueMetrics) {
-		updateLocalQueueResourceMetrics(newLq)
-	}
```

Comment:
@mimowo Here is the fix. This block will execute only if the StopPolicy has changed. This is not correct. We should update metrics after each update.